From 1211a10b959fea039f4c8718c3901ad63e4816d3 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 2 Jul 2015 15:40:13 +0200
Subject: [PATCH] splitup dracut-init.sh from dracut-functions.sh

other tools want to source dracut-functions.sh without any mkdir side
effects.

(cherry picked from commit 777f2db0373ccbc1a44fc2d960ecefbe50195055)
---
 Makefile                          |  1 +
 dracut-functions.sh               |  9 ++-------
 dracut-init.sh                    | 34 ++++++++++++++++++++++++++++++++++
 dracut.sh                         |  8 ++++----
 dracut.spec                       |  1 +
 test/TEST-01-BASIC/test.sh        |  6 +++---
 test/TEST-02-SYSTEMD/test.sh      |  6 +++---
 test/TEST-03-USR-MOUNT/test.sh    |  6 +++---
 test/TEST-04-FULL-SYSTEMD/test.sh |  6 +++---
 test/TEST-10-RAID/test.sh         |  6 +++---
 test/TEST-11-LVM/test.sh          |  6 +++---
 test/TEST-12-RAID-DEG/test.sh     |  6 +++---
 test/TEST-13-ENC-RAID-LVM/test.sh |  6 +++---
 test/TEST-14-IMSM/test.sh         |  6 +++---
 test/TEST-15-BTRFSRAID/test.sh    |  6 +++---
 test/TEST-16-DMSQUASH/test.sh     |  4 ++--
 test/TEST-17-LVM-THIN/test.sh     |  6 +++---
 test/TEST-20-NFS/test.sh          |  6 +++---
 test/TEST-30-ISCSI/test.sh        |  8 ++++----
 test/TEST-40-NBD/test.sh          | 10 +++++-----
 test/TEST-50-MULTINIC/test.sh     |  6 +++---
 21 files changed, 92 insertions(+), 61 deletions(-)
 create mode 100644 dracut-init.sh

diff --git a/Makefile b/Makefile
index 5441d8f..18f5a88 100644
--- a/Makefile
+++ b/Makefile
@@ -106,6 +106,7 @@ install: dracut-version.sh
 	install -m 0644 dracut.conf $(DESTDIR)$(sysconfdir)/dracut.conf
 	mkdir -p $(DESTDIR)$(sysconfdir)/dracut.conf.d
 	mkdir -p $(DESTDIR)$(pkglibdir)/dracut.conf.d
+	install -m 0755 dracut-init.sh $(DESTDIR)$(pkglibdir)/dracut-init.sh
 	install -m 0755 dracut-functions.sh $(DESTDIR)$(pkglibdir)/dracut-functions.sh
 	install -m 0755 dracut-version.sh $(DESTDIR)$(pkglibdir)/dracut-version.sh
 	ln -fs dracut-functions.sh $(DESTDIR)$(pkglibdir)/dracut-functions
diff --git a/dracut-functions.sh b/dracut-functions.sh
index 4ea3204..31586ca 100755
--- a/dracut-functions.sh
+++ b/dracut-functions.sh
@@ -22,14 +22,9 @@
 export LC_MESSAGES=C
 
 if [[ $DRACUT_KERNEL_LAZY ]] && ! [[ $DRACUT_KERNEL_LAZY_HASHDIR ]]; then
-    if ! [[ -d "$initdir/.kernelmodseen" ]]; then
-        mkdir -p "$initdir/.kernelmodseen"
+    if [[ -d "$initdir/.kernelmodseen" ]]; then
+        DRACUT_KERNEL_LAZY_HASHDIR="$initdir/.kernelmodseen"
     fi
-    DRACUT_KERNEL_LAZY_HASHDIR="$initdir/.kernelmodseen"
-fi
-
-if [[ $initdir ]] && ! [[ -d $initdir ]]; then
-    mkdir -p "$initdir"
 fi
 
 # Generic substring function.  If $2 is in $1, return 0.
diff --git a/dracut-init.sh b/dracut-init.sh
new file mode 100644
index 0000000..469e42a
--- /dev/null
+++ b/dracut-init.sh
@@ -0,0 +1,34 @@
+#!/bin/bash
+#
+# functions used by dracut and other tools.
+#
+# Copyright 2005-2009 Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+export LC_MESSAGES=C
+
+if [[ $DRACUT_KERNEL_LAZY ]] && ! [[ $DRACUT_KERNEL_LAZY_HASHDIR ]]; then
+    if ! [[ -d "$initdir/.kernelmodseen" ]]; then
+        mkdir -p "$initdir/.kernelmodseen"
+    fi
+    DRACUT_KERNEL_LAZY_HASHDIR="$initdir/.kernelmodseen"
+fi
+
+if [[ $initdir ]] && ! [[ -d $initdir ]]; then
+    mkdir -p "$initdir"
+fi
+
+[[ $dracutbasedir ]] || export dracutbasedir=${BASH_SOURCE%/*}
+. $dracutbasedir/dracut-functions.sh
diff --git a/dracut.sh b/dracut.sh
index cc22a14..a214f81 100755
--- a/dracut.sh
+++ b/dracut.sh
@@ -519,7 +519,7 @@ while :; do
         -q|--quiet)    ((verbosity_mod_l--));;
         -l|--local)
                        allowlocal="yes"
-                       [[ -f "$(readlink -f "${0%/*}")/dracut-functions.sh" ]] \
+                       [[ -f "$(readlink -f "${0%/*}")/dracut-init.sh" ]] \
                            && dracutbasedir="$(readlink -f "${0%/*}")"
                        ;;
         -H|--hostonly|--host-only)
@@ -841,10 +841,10 @@ if [[ $print_cmdline ]]; then
     kmsgloglvl=0
 fi
 
-if [[ -f $dracutbasedir/dracut-functions.sh ]]; then
-    . $dracutbasedir/dracut-functions.sh
+if [[ -f $dracutbasedir/dracut-init.sh ]]; then
+    . $dracutbasedir/dracut-init.sh
 else
-    printf "%s\n" "dracut: Cannot find $dracutbasedir/dracut-functions.sh." >&2
+    printf "%s\n" "dracut: Cannot find $dracutbasedir/dracut-init.sh." >&2
     printf "%s\n" "dracut: Are you running from a git checkout?" >&2
     printf "%s\n" "dracut: Try passing -l as an argument to $0" >&2
     exit 1
diff --git a/dracut.spec b/dracut.spec
index 302cfc5..d22f586 100644
--- a/dracut.spec
+++ b/dracut.spec
@@ -296,6 +296,7 @@ rm -rf -- $RPM_BUILD_ROOT
 %endif
 %dir %{dracutlibdir}
 %dir %{dracutlibdir}/modules.d
+%{dracutlibdir}/dracut-init.sh
 %{dracutlibdir}/dracut-functions.sh
 %{dracutlibdir}/dracut-functions
 %{dracutlibdir}/dracut-version.sh
diff --git a/test/TEST-01-BASIC/test.sh b/test/TEST-01-BASIC/test.sh
index 02ebc23..642b250 100755
--- a/test/TEST-01-BASIC/test.sh
+++ b/test/TEST-01-BASIC/test.sh
@@ -29,7 +29,7 @@ test_setup() {
     (
 	export initdir=$TESTDIR/overlay/source
 	mkdir -p $initdir
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient \
 	    umount strace less setsid
@@ -51,7 +51,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mkfs.ext3 poweroff cp umount sync
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -80,7 +80,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-02-SYSTEMD/test.sh b/test/TEST-02-SYSTEMD/test.sh
index 2a8ed6b..4ebff83 100755
--- a/test/TEST-02-SYSTEMD/test.sh
+++ b/test/TEST-02-SYSTEMD/test.sh
@@ -25,7 +25,7 @@ test_setup() {
     (
 	export initdir=$TESTDIR/overlay/source
 	mkdir -p $initdir
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient \
 	    umount strace less setsid
@@ -47,7 +47,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mkfs.ext3 poweroff cp umount
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -76,7 +76,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-03-USR-MOUNT/test.sh b/test/TEST-03-USR-MOUNT/test.sh
index b8231fe..e772352 100755
--- a/test/TEST-03-USR-MOUNT/test.sh
+++ b/test/TEST-03-USR-MOUNT/test.sh
@@ -55,7 +55,7 @@ test_setup() {
     (
 	export initdir=$TESTDIR/overlay/source
 	mkdir -p $initdir
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient \
 	    umount strace less setsid
@@ -78,7 +78,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mkfs.btrfs btrfs poweroff cp umount sync
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -114,7 +114,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-04-FULL-SYSTEMD/test.sh b/test/TEST-04-FULL-SYSTEMD/test.sh
index aa83122..3834238 100755
--- a/test/TEST-04-FULL-SYSTEMD/test.sh
+++ b/test/TEST-04-FULL-SYSTEMD/test.sh
@@ -57,7 +57,7 @@ test_setup() {
     (
 	export initdir=$TESTDIR/overlay/source
 	mkdir -p $initdir
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 
         for d in usr/bin usr/sbin bin etc lib "$libdir" sbin tmp usr var var/log dev proc sys sysroot root run; do
             if [ -L "/$d" ]; then
@@ -221,7 +221,7 @@ EOF
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mkfs.btrfs btrfs poweroff cp umount sync
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -256,7 +256,7 @@ EOF
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-10-RAID/test.sh b/test/TEST-10-RAID/test.sh
index d79d5cd..cf474df 100755
--- a/test/TEST-10-RAID/test.sh
+++ b/test/TEST-10-RAID/test.sh
@@ -28,7 +28,7 @@ test_setup() {
     (
 	export initdir=$TESTDIR/overlay/source
 	(mkdir -p "$initdir"; cd "$initdir"; mkdir -p dev sys proc etc var/run tmp run)
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -49,7 +49,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -77,7 +77,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst ./cryptroot-ask.sh /sbin/cryptroot-ask
diff --git a/test/TEST-11-LVM/test.sh b/test/TEST-11-LVM/test.sh
index 732d0c8..c3c27bc 100755
--- a/test/TEST-11-LVM/test.sh
+++ b/test/TEST-11-LVM/test.sh
@@ -24,7 +24,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -46,7 +46,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -69,7 +69,7 @@ test_setup() {
     grep -F -m 1 -q dracut-root-block-created $TESTDIR/root.ext2 || return 1
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-12-RAID-DEG/test.sh b/test/TEST-12-RAID-DEG/test.sh
index 87fe542..11f66b2 100755
--- a/test/TEST-12-RAID-DEG/test.sh
+++ b/test/TEST-12-RAID-DEG/test.sh
@@ -66,7 +66,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -87,7 +87,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount dd grep
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -120,7 +120,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-13-ENC-RAID-LVM/test.sh b/test/TEST-13-ENC-RAID-LVM/test.sh
index 129620b..993f71d 100755
--- a/test/TEST-13-ENC-RAID-LVM/test.sh
+++ b/test/TEST-13-ENC-RAID-LVM/test.sh
@@ -62,7 +62,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -83,7 +83,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount grep
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -113,7 +113,7 @@ test_setup() {
 
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-14-IMSM/test.sh b/test/TEST-14-IMSM/test.sh
index ad0cf7b..9465965 100755
--- a/test/TEST-14-IMSM/test.sh
+++ b/test/TEST-14-IMSM/test.sh
@@ -54,7 +54,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -76,7 +76,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount grep
 	inst_hook initqueue 01 ./create-root.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
@@ -104,7 +104,7 @@ test_setup() {
     echo $MD_UUID > $TESTDIR/mduuid
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-15-BTRFSRAID/test.sh b/test/TEST-15-BTRFSRAID/test.sh
index 71f5703..9b97136 100755
--- a/test/TEST-15-BTRFSRAID/test.sh
+++ b/test/TEST-15-BTRFSRAID/test.sh
@@ -26,7 +26,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -47,7 +47,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mkfs.btrfs poweroff cp umount
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -77,7 +77,7 @@ test_setup() {
 
    (
         export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-16-DMSQUASH/test.sh b/test/TEST-16-DMSQUASH/test.sh
index 252e845..c08a2a4 100755
--- a/test/TEST-16-DMSQUASH/test.sh
+++ b/test/TEST-16-DMSQUASH/test.sh
@@ -30,7 +30,7 @@ test_setup() {
     mkdir -p -- "$TESTDIR"/overlay
     (
 	export initdir="$TESTDIR"/overlay
-	. "$basedir"/dracut-functions.sh
+	. "$basedir"/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
@@ -48,7 +48,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir="$TESTDIR"/root-source
-	. "$basedir"/dracut-functions.sh
+	. "$basedir"/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient \
 	    umount strace less
diff --git a/test/TEST-17-LVM-THIN/test.sh b/test/TEST-17-LVM-THIN/test.sh
index 4c904b0..79e224e 100755
--- a/test/TEST-17-LVM-THIN/test.sh
+++ b/test/TEST-17-LVM-THIN/test.sh
@@ -24,7 +24,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
 	export initdir=$TESTDIR/overlay/source
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip route \
 	    mount dmesg ifconfig dhclient mkdir cp ping dhclient
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -46,7 +46,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple sfdisk mke2fs poweroff cp umount
 	inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -69,7 +69,7 @@ test_setup() {
     grep -F -m 1 -q dracut-root-block-created $TESTDIR/root.ext2 || return 1
     (
 	export initdir=$TESTDIR/overlay
-	. $basedir/dracut-functions.sh
+	. $basedir/dracut-init.sh
 	inst_multiple poweroff shutdown
 	inst_hook emergency 000 ./hard-off.sh
 	inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-20-NFS/test.sh b/test/TEST-20-NFS/test.sh
index 7b5e3c5..14350e9 100755
--- a/test/TEST-20-NFS/test.sh
+++ b/test/TEST-20-NFS/test.sh
@@ -221,7 +221,7 @@ test_setup() {
 
     (
         export initdir=$TESTDIR/mnt
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
 
         for _f in modules.builtin.bin modules.builtin; do
             [[ $srcmods/$_f ]] && break
@@ -287,7 +287,7 @@ test_setup() {
     # Make client root inside server root
     (
         export initdir=$TESTDIR/mnt/nfs/client
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
 
         inst_multiple sh shutdown poweroff stty cat ps ln ip \
             mount dmesg mkdir cp ping grep
@@ -331,7 +331,7 @@ test_setup() {
     # Make an overlay with needed tools for the test harness
     (
         export initdir=$TESTDIR/overlay
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         mkdir $TESTDIR/overlay
         inst_multiple poweroff shutdown
         inst_hook emergency 000 ./hard-off.sh
diff --git a/test/TEST-30-ISCSI/test.sh b/test/TEST-30-ISCSI/test.sh
index f597783..a258c04 100755
--- a/test/TEST-30-ISCSI/test.sh
+++ b/test/TEST-30-ISCSI/test.sh
@@ -104,7 +104,7 @@ test_setup() {
     # Create what will eventually be our root filesystem onto an overlay
     (
         export initdir=$TESTDIR/overlay/source
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         inst_multiple sh shutdown poweroff stty cat ps ln ip \
             mount dmesg mkdir cp ping grep
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -121,7 +121,7 @@ test_setup() {
     # second, install the files needed to make the root filesystem
     (
         export initdir=$TESTDIR/overlay
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         inst_multiple sfdisk mkfs.ext3 poweroff cp umount
         inst_hook initqueue 01 ./create-root.sh
         inst_hook initqueue/finished 01 ./finished-false.sh
@@ -157,7 +157,7 @@ test_setup() {
     rm -- $TESTDIR/client.img
     (
         export initdir=$TESTDIR/overlay
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         inst_multiple poweroff shutdown
         inst_hook emergency 000 ./hard-off.sh
         inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
@@ -177,7 +177,7 @@ test_setup() {
     kernel=$KVERSION
     (
         export initdir=$TESTDIR/mnt
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         (
             cd "$initdir";
             mkdir -p dev sys proc etc var/run tmp var/lib/dhcpd /etc/iscsi
diff --git a/test/TEST-40-NBD/test.sh b/test/TEST-40-NBD/test.sh
index 59494ed..abf1a4f 100755
--- a/test/TEST-40-NBD/test.sh
+++ b/test/TEST-40-NBD/test.sh
@@ -195,7 +195,7 @@ make_encrypted_root() {
     # Create what will eventually be our root filesystem onto an overlay
     (
         export initdir=$TESTDIR/overlay/source
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         mkdir -p "$initdir"
         (cd "$initdir"; mkdir -p dev sys proc etc var/run tmp )
         inst_multiple sh df free ls shutdown poweroff stty cat ps ln ip \
@@ -214,7 +214,7 @@ make_encrypted_root() {
     # second, install the files needed to make the root filesystem
     (
         export initdir=$TESTDIR/overlay
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         inst_multiple mke2fs poweroff cp umount tune2fs
         inst_hook emergency 000 ./hard-off.sh
         inst_hook initqueue 01 ./create-root.sh
@@ -253,7 +253,7 @@ make_client_root() {
     kernel=$KVERSION
     (
         export initdir=$TESTDIR/mnt
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         mkdir -p "$initdir"
         (cd "$initdir"; mkdir -p dev sys proc etc var/run tmp )
         inst_multiple sh ls shutdown poweroff stty cat ps ln ip \
@@ -288,7 +288,7 @@ make_server_root() {
     kernel=$KVERSION
     (
         export initdir=$TESTDIR/mnt
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         mkdir -p "$initdir"
         (
             cd "$initdir";
@@ -335,7 +335,7 @@ test_setup() {
     # Make the test image
     (
         export initdir=$TESTDIR/overlay
-        . $basedir/dracut-functions.sh
+        . $basedir/dracut-init.sh
         inst_multiple poweroff shutdown
         inst_hook emergency 000 ./hard-off.sh
         inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
diff --git a/test/TEST-50-MULTINIC/test.sh b/test/TEST-50-MULTINIC/test.sh
index 3c6003b..4f73203 100755
--- a/test/TEST-50-MULTINIC/test.sh
+++ b/test/TEST-50-MULTINIC/test.sh
@@ -150,7 +150,7 @@ test_setup() {
 
     (
         export initdir="$TESTDIR"/mnt
-        . "$basedir"/dracut-functions.sh
+        . "$basedir"/dracut-init.sh
 
         (
             cd "$initdir";
@@ -216,7 +216,7 @@ test_setup() {
     # Make client root inside server root
     (
         export initdir="$TESTDIR"/mnt/nfs/client
-        . "$basedir"/dracut-functions.sh
+        . "$basedir"/dracut-init.sh
         inst_multiple sh shutdown poweroff stty cat ps ln ip \
             mount dmesg mkdir cp ping grep ls
         for _terminfodir in /lib/terminfo /etc/terminfo /usr/share/terminfo; do
@@ -256,7 +256,7 @@ test_setup() {
     # Make an overlay with needed tools for the test harness
     (
         export initdir="$TESTDIR"/overlay
-        . "$basedir"/dracut-functions.sh
+        . "$basedir"/dracut-init.sh
         inst_multiple poweroff shutdown
         inst_hook emergency 000 ./hard-off.sh
         inst_simple ./99-idesymlinks.rules /etc/udev/rules.d/99-idesymlinks.rules
